daml-lf/engine/src/main/scala/com/digitalasset/daml/lf/engine/Error.scala

```diff
@@ -127,6 +127,12 @@ object Error {
 
     final case class BadDisclosedContract(message: String) extends Error
 
+    final case class AmbiguousInterfaceInstance(interface: Ref.Identifier, template: Ref.Identifier)
+        extends Error {
+      override def message: String =
+        s"Ambiguous interface instance: two instances for interface $interface and template $template"
+    }
+
```

Comment:
I would have define the  AmbiguousInterfaceInstance as a new kind of `LookupError`. Something like that:


sealed abstract class LookupError {
  def pretty: String
}

```
object LookupError {

  final case class NotFound(notFound: Reference, context: Reference) extends LookupError {
    def pretty: String = "unknown " + notFound.pretty + (
      if (context == notFound) "" else LookupError.contextDetails(context)
    )
  }

  final case class AmbiguousInterfaceInstance(instance: Reference.InterfaceInstance)
      extends LookupError {
    def pretty: String = s"Ambiguous interface instance: two instances for ${instance.pretty}"

  }

 [...]

  object MissingPackage {
    def unapply(err: LookupError.NotFound): Option[(PackageId, Reference)] = {
      err.notFound match {
        case Reference.Package(packageId) => Some(packageId -> err.context)
        case _ => None
      }
    }
```

Reply:
- I do not want to expose a new error on the ledger API
- This is not really a engine error, but a lookup error, i.e. an issue in the dars, this could theoretically happen in other pieces of code.

Reply:
I'm not sure this error makes sense at all. This should be caught during typechecking. If you get this when computing an interface view your typechecker is broken. I'd put this under an internal error.

Reply:
@cocreature is right. 
This should be an internal error, not reason to expose it to the user.

Reply:
@cocreature is right, this error should not happened for properly typed package. 
The simpler is to throw `Error.Preprocessing.Internal`

Reply:
> I would have defined the `AmbiguousInterfaceInstance` as a new kind of `LookupError`
> -- @remyhaemmerle-da 

Ah yes, one more data point :) Since I introduced `AmbiguousInterfaceInstance`, I'll take care of this in a separate PR.

> @cocreature is right, this error should not happened for properly typed package
> The simpler is to throw `Error.Preprocessing.Internal`
> -- @remyhaemmerle-da 

I agree here too, so for this PR let's just throw an internal error - once I make `AmbiguousInterfaceInstance` a `LookupError`, the code in `CommandPreprocessor` should become a `handleLookup` one-liner

Reply:
> I agree here too, so for this PR let's just throw an internal error - once I make AmbiguousInterfaceInstance a LookupError, the code in CommandPreprocessor should become a handleLookup one-liner

Just made these changes since I didn't see these updates to the conversation

Reply:
Thank you for fixing that @dylant-da. Please keep those changes in this PR, no need to add extra work.